### ISSUE TITLE
some confusion about cudnn batch norm

### ISSUE THREAD

**Issue by wranglerwong** [apache/incubator-mxnet#3091]
@piiswrong hey, I am trying to finetune a pretrained Residual Network model.
And I define my symbol as 
`bn = mx.symbol.BatchNorm(data=data, name="bn")`

And I am using CUDA 8.0 + CUDNN v5 + GTX1080, as the codes in batch_norm.cu
`#if MXNET_USE_CUDNN == 1 && CUDNN_MAJOR >= 5
  return new CuDNNBatchNormOp(param);
# else

  return new BatchNormOp<gpu>(param);
# endif`

It will use **cudnn bn** by default, so I am just wondering how it can acquire the **moving_inv_var** in cudnn_batch_norm-inl.h because there are only **moving_var** params in the pretrained model, and after serveral epochs, there are still moving_var but not the moving_inv_var params saved in new finetuned models. 
Could you explain to me a little bit how this works. Thanks.

**Comment by winstywang** [comment]
You have to use the non-cudnn version of BN if you want to use old pretrained model. Try to return new BatchNormOp<gpu>(param); directly.

**Comment by wranglerwong** [comment]
@winstywang Thanks for your answer.
I have tried both. Using cudnn version of BN is ok while using the non-cudnn BN seems can't converge. And there are no **moving_inv_var** params in models trained by using cudnn BN.
How could these be?
Thank you so much!!

**Comment by winstywang** [comment]
I see your question. This is caused by the behavior change of BN layer from cudnn v4 to v5. In v4, it uses inv var, however in v5 it changes to var. So the moving_inv_var is only for cudnn v4, it is not used in current batch_norm.cu

**Comment by wranglerwong** [comment]
@winstywang Thank you so much!!

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
